frontend/src/processTubeData.js

```diff
@@ -3,6 +3,90 @@
 const bpm = 60; 
 const noteInterval = 60 / (bpm * 4); // for 1/16th note in seconds
 const randomIntervalMultiplier = 1 / noteInterval;
+const spreadNewData = true; // if true, spread new data across timeToStation values
+
+// separate the data into an array of objects for each train line
+function separateDataIntoLines(tubeData) {
+  if (tubeData.length > 0) {
+    const separatedData = tubeData.reduce((acc, obj) => { // acc = accumulator
+      // the accumulator is an object with keys for each lineName
+      const lineName = obj.lineName; 
+      if (!acc[lineName]) {
+        acc[lineName] = [];
+      } 
+      acc[lineName].push(obj);
+      return acc;
+    }, {});
+    return separatedData;
+  }
+}
+
+// declare the fudgeData function to iterate through each array in separatedData calling spreadData on each array and return the result
+function fudgeData(separatedData, dataBlockDuration) {
+  if (Object.keys(separatedData).length > 0) {
+    const fudgedDataArray = [];
+    Object.values(separatedData).forEach((array) => {
+      const result = spreadData(array, dataBlockDuration);
+      fudgedDataArray.push(result);
+    });
+    const fudgedData = fudgedDataArray.flat();
+    return fudgedData;
+  } else {
+    console.log("No data");
+  }
+}
+
+// The spreadData function looks at the timeToStation of each object in the array and copies some objects to other timeToStation values where there are more than three objects within a second interval
+function spreadData(array, dataBlockDuration) {
```

Comment:
This sounds great, thank you for doing this, it definitely helps. I wonder if we could shorten this function, and I am finding it a little difficult to read/ follow, but that may just be my bad javascript skills!